recipes/gz-sim/build_cxx.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+if [[ "${target_platform}" == osx-* ]]; then
+    # See https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
+    CXXFLAGS="${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
+fi
+
+if [[ "${CONDA_BUILD_CROSS_COMPILATION}" == "1" ]]; then
+  export CMAKE_ARGS="${CMAKE_ARGS} -DProtobuf_PROTOC_EXECUTABLE=$BUILD_PREFIX/bin/protoc"
+fi
+
+mkdir build
+cd build
+
+cmake ${CMAKE_ARGS} -GNinja .. \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=True \
```

Comment:
We probably don't need this in linux:
```
2022-10-16T20:14:15.0348427Z CMake Warning:
2022-10-16T20:14:15.0351812Z   Manually-specified variables were not used by the project:
2022-10-16T20:14:15.0354384Z 
2022-10-16T20:14:15.0356888Z     CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP
```

Reply:
Thanks, done in https://github.com/conda-forge/staged-recipes/pull/20686/commits/3cafedcec82fe37c2b9d8e74e35fbcc87df13c41 .